dbt/adapters/base/impl.py

```diff
@@ -1743,6 +1746,15 @@ def capabilities(cls) -> CapabilityDict:
     def supports(cls, capability: Capability) -> bool:
         return bool(cls.capabilities()[capability])
 
+    @available
+    @classmethod
+    def get_adapter_run_info(cls, config: RelationConfig) -> AdapterTrackingRelationInfo:
+        return AdapterTrackingRelationInfo(
```

Comment:
I could see wanting to return both the `BaseAdapter` version as well as the concrete adapter version. Can we add that in? It's also possible to predict both `adapter_name` and the path to the adapter version (`dbt.adapters.<adapter_name>.__version__.version`) as we do in `dbt.adapters.factory`. Could we fill that in for concrete adapters? And how would we extend this to combine the two? For example, if we had this method plus an adapter specific one, then we could merge them together:

```python
@classmethod
def get_adapter_run_info(cls, config) -> AdapterTrackingRelationInfo:
    """
    Interface method for collecting adapter metadata for tracking. This method should not be overwritten; instead, `self._get_adapter_run_info()` should be overwritten.
    """
    return AdapterTrackingRelationInfo(
        adapter_name=self.type,
        adapter_version=import_module(f"dbt.adapters.dbt_{self.type}.__version__).version,
        base_adapter_version=import_module(f"dbt.adapters.__about__).version,
        adapter_details=self._get_adapter_run_info(config),
    )

@classmethod
def _get_adapter_run_info(cls, config) -> Dict[str, Any]:
    """
    Adapter maintainers should overwrite this method to return any run metadata that should be captured during a run.
    """
    return {}
```